src/js/core/factories/GridScrolling.js

```diff
@@ -0,0 +1,300 @@
+(function() {
+  'use strict';
+
+  angular.module('ui.grid')
+    .factory('gridScrolling', ['$window', 'gridUtil', 'uiGridConstants',
+      function($window, gridUtil, uiGridConstants) {
+        var isAnimating;
+
+        /**
+         *  @ngdoc object
+         *  @name initiated
+         *  @propertyOf  ui.grid.class:gridScrolling
+         *  @description Keeps track of which type of scrolling event has been initiated
+         *  and sets it to NONE, when no event is being triggered.
+         */
+        gridScrolling.initiated = uiGridConstants.scrollType.NONE;
+
+        /**
+         * @ngdoc function
+         * @name ui.grid.class:gridScrolling
+         * @description gridScrolling is a wrapper service that takes over the default scrolling logic in order to
+         * ensure that grid scrolling works consistently in both the browser and devices, as well as slow machines.
+         * @param {object} element Element being scrolled
+         * @param {function} scrollHandler Function that needs to be called when scrolling happens.
+         */
+        function gridScrolling(element, scrollHandler) {
+          var wrapper = element, pointX, pointY, startTime, startX, startY, maxScroll,
+            scroller = wrapper[0].children[0],
+            initType = {
+              touchstart: uiGridConstants.scrollType.TOUCHABLE,
+              touchmove: uiGridConstants.scrollType.TOUCHABLE,
+              touchend: uiGridConstants.scrollType.TOUCHABLE,
+
+              mousedown: uiGridConstants.scrollType.MOUSE,
+              mousemove: uiGridConstants.scrollType.MOUSE,
+              mouseup: uiGridConstants.scrollType.MOUSE,
+
+              pointerdown: uiGridConstants.scrollType.POINTER,
+              pointermove: uiGridConstants.scrollType.POINTER,
+              pointerup: uiGridConstants.scrollType.POINTER
+            };
+
+          if ('onmousedown' in $window) {
+            wrapper.on('scroll', scrollHandler);
+          }
+
+          if (gridUtil.isTouchEnabled()) {
+            wrapper.on('touchstart', start);
+            wrapper.on('touchmove', move);
+            wrapper.on('touchcancel', end);
+            wrapper.on('touchend', end);
+            document.addEventListener('touchmove', function(e) {
+              e.preventDefault();
+            }, false);
+          }
+
+          /**
+           * @ngdoc function
+           * @name start
+           * @methodOf ui.grid.class:gridScrolling
+           * @description Gets the current coordinates and time, as well as the target coordinate
+           * and initializes the scrolling event
+           * @param {object} event The event object
+           */
+          function start(event) {
+            var point = event.touches ? event.touches[0] : event;
+
+            wrapper.off('scroll', scrollHandler);
```

Comment:
to understand this correctly, you are turning off native scrolling here and doing it yourself in this service?